src/Storages/tests/gtest_azure_configuration.cpp

```diff
@@ -0,0 +1,183 @@
+#include <Poco/DOM/DOMParser.h>
+
+using namespace DB;
+
+namespace
+{
+
+/// A class which allows to test private methods of NamedCollectionFactory.
+class NamedCollectionFactoryFriend : public NamedCollectionFactory
+{
+public:
+    static NamedCollectionFactoryFriend & instance()
+    {
+        static NamedCollectionFactoryFriend instance;
+        return instance;
+    }
+
+    void loadFromConfig(const Poco::Util::AbstractConfiguration & config)
+    {
+        std::lock_guard lock(mutex);
+        NamedCollectionFactory::loadFromConfig(config, lock);
+    }
+};
+
+class StorageAzureConfigurationFriend : public StorageAzureConfiguration
+{
+public:
+    void fromNamedCollection(const NamedCollection & collection, ContextPtr context) override
+    {
+        StorageAzureConfiguration::fromNamedCollection(collection, context);
+    }
+
+    void fromAST(ASTs & args, ContextPtr context, bool with_structure) override
+    {
+        StorageAzureConfiguration::fromAST(args, context, with_structure);
+    }
+
+    const AzureBlobStorage::ConnectionParams & getConnectionParams()
+    {
+        return connection_params;
+    }
+};
+
+void loadNamedCollectionConfig(const String & xml)
+{
+    Poco::XML::DOMParser dom_parser;
+    Poco::AutoPtr<Poco::XML::Document> document = dom_parser.parseString(xml);
+    Poco::AutoPtr<Poco::Util::XMLConfiguration> config = new Poco::Util::XMLConfiguration(document);
+
+    NamedCollectionFactoryFriend::instance().loadFromConfig(*config);
+}
+
+TEST(StorageAzureConfiguration, FromNamedCollectionWithExtraCredentials)
+{
+    std::string xml(R"CONFIG(<clickhouse>
+    <named_collections>
+        <FromNamedCollectionWithExtraCredentials>
+            <container>test_container</container>
+            <blob_path>test_blob.csv</blob_path>
+            <client_id>test_client_id</client_id>
+            <tenant_id>test_tenant_id</tenant_id>
+        </FromNamedCollectionWithExtraCredentials>
+    </named_collections>
+    </clickhouse>)CONFIG");
+
+    loadNamedCollectionConfig(xml);
+
+    StorageAzureConfigurationFriend conf;
+    auto collection = NamedCollectionFactoryFriend::instance().get("FromNamedCollectionWithExtraCredentials");
+    conf.fromNamedCollection(*collection, Context::getGlobalContextInstance());
+
+    ASSERT_TRUE(std::holds_alternative<std::shared_ptr<Azure::Identity::WorkloadIdentityCredential>>(conf.getConnectionParams().auth_method));
+}
+
+TEST(StorageAzureConfiguration, FromNamedCollectionWithExtraCredentialsAndAccount)
+{
+    std::string xml(R"CONFIG(<clickhouse>
+    <named_collections>
+        <FromNamedCollectionWithExtraCredentialsAndAccount>
+            <container>test_container</container>
+            <blob_path>test_blob.csv</blob_path>
+            <account_name>test_account</account_name>
+            <account_key>test_key</account_key>
+            <client_id>test_client_id</client_id>
+            <tenant_id>test_tenant_id</tenant_id>
```

Comment:
In cases where both authentication methods are provided, maybe we can throw an error saying you provided both or if we are sure we always want to prioritise WorkloadIdentity then we should add this to the doc and also add a log that we are using WorkloadIdentity or the other one. Please let me know your thoughts on this